src/sentry/dynamic_sampling/utils.py

```diff
@@ -43,3 +54,54 @@ def generate_uniform_rule(project: Project) -> UniformRule:
         },
         "id": UNIFORM_RULE_RESERVED_ID,
     }
+
+
+class DynamicSamplingFeatureMultiplexer:
+    """
+    This class is used to route DS behaviour according to the feature flags listed here
+    Essentially the logic is as follows:
+    - The `organizations:server-side-sampling` feature flag is the main flag enabled for dynamic sampling both in
+    sentry and in relay
+    - The  `organizations:server-side-sampling-ui` feature flag is the flag that enables the old dynamic sampling
+    behaviour which needs to be supported for backwards compatibility but is deprecated
+    - The `organizations:dynamic-sampling-basic` feature flag is the flag that enables the new adaptive sampling
+    """
+
+    def __init__(self, project, request):
+        # Feature flag that informs us that relay is handling DS rules
+        self.allow_dynamic_sampling = features.has(
+            "organizations:server-side-sampling", project.organization, actor=request.user
+        )
+        # Feature flag that informs us that the org is on the new AM2 plan and thereby have adaptive sampling enabled
+        # ToDo(ahmed): This needs to be renamed to `organizations:dynamic-sampling`
+        self.current_dynamic_sampling = features.has(
```

Comment:
I wouldn't call it current and I would use the `is` form to make the boolean nature implicit. My proposal would be `is_dynamic_sampling_basic`.